oxd-server/src/main/java/org/gluu/oxd/server/mapper/RegisterRequestMapper.java

```diff
@@ -0,0 +1,410 @@
+            rp.setAuthorizedOrigins(request.getAuthorizedOrigins());
+        }
+
+        if (request.getAccessTokenLifetime() != null && NumberUtils.isNumber(request.getAccessTokenLifetime().toString())) {
+            rp.setAccessTokenLifetime(request.getAccessTokenLifetime());
+        }
+
+        if (!Strings.isNullOrEmpty(request.getSoftwareId())) {
+            rp.setSoftwareId(request.getSoftwareId());
+        }
+
+        if (!Strings.isNullOrEmpty(request.getSoftwareVersion())) {
+            rp.setSoftwareVersion(request.getSoftwareVersion());
+        }
+
+        if (!Strings.isNullOrEmpty(request.getSoftwareStatement())) {
+            rp.setSoftwareStatement(request.getSoftwareStatement());
+        }
+
+        if (request.getCustomAttributes() != null && !request.getCustomAttributes().isEmpty()) {
+            request.getCustomAttributes().entrySet().stream().forEach(e -> {
+                rp.addCustomAttribute(e.getKey(), e.getValue());
+            });
+        }
+
+        if (!Strings.isNullOrEmpty(request.getJwksUri())) {
+            rp.setClientJwksUri(request.getJwksUri());
+        }
+
+        if (request.getClaimsRedirectUris() != null && !request.getClaimsRedirectUris().isEmpty()) {
+            rp.setClaimsRedirectUri(request.getClaimsRedirectUris());
+        }
+
+    }
+
+    public RegisterRequest createRegisterRequest(Rp rp) {
+        final RegisterRequest request = new RegisterRequest(rp.getClientRegistrationAccessToken());
+
+        if (!Strings.isNullOrEmpty(rp.getClientName())) {
+            request.setClientName(rp.getClientName());
+        }
+
+        if (!Strings.isNullOrEmpty(rp.getApplicationType())) {
+            request.setApplicationType(ApplicationType.fromString(rp.getApplicationType()));
+        }
+
+        if (!Strings.isNullOrEmpty(rp.getTokenEndpointAuthSigningAlg())) {
+            request.setTokenEndpointAuthSigningAlg(SignatureAlgorithm.fromString(rp.getTokenEndpointAuthSigningAlg()));
+        }
+
+        if (rp.getGrantType() != null) {
+            request.setGrantTypes(rp.getGrantType().stream().map(item -> GrantType.fromString(item)).collect(Collectors.toList()));
+        }
+
+        if (rp.getFrontChannelLogoutUris() != null) {
+            request.setFrontChannelLogoutUris(rp.getFrontChannelLogoutUris());
+        }
+
+        if (!Strings.isNullOrEmpty(rp.getTokenEndpointAuthMethod())) {
+            request.setTokenEndpointAuthMethod(AuthenticationMethod.fromString(rp.getTokenEndpointAuthMethod()));
+        }
+
+        if (rp.getClientRequestUris() != null && !rp.getClientRequestUris().isEmpty()) {
+            request.setRequestUris(rp.getClientRequestUris());
+        }
+
+        if (!Strings.isNullOrEmpty(rp.getClientSectorIdentifierUri())) {
+            request.setSectorIdentifierUri(rp.getClientSectorIdentifierUri());
+        }
+
+        if (rp.getRedirectUris() != null && !rp.getRedirectUris().isEmpty()) {
```

Comment:
It time for `org.apache.commons.collections.CollectionUtils#isEmpty`. Add static import.

Reply:
Changes done